crates/subspace-core-primitives/Cargo.toml

```diff
@@ -44,19 +44,25 @@ criterion = "0.4.0"
 [features]
 default = [
     "parallel-decoding",
+    "serde",
     "std",
 ]
 # Parallel decoding will use all CPUs available, but will allocate a memory of a size of a sector instead of square root
 # of that
 parallel-decoding = ["rayon"]
+serde = [
```

Comment:
There is no advantage of defining a separate `serde` feature here. Enabling only `serde` results in compiler error as code here https://github.com/subspace/subspace/blob/3e35060edbc56d2b4fce0e8c817c72fc4a47a0d6/crates/subspace-core-primitives/src/crypto/kzg/serde.rs#L23

assumes that by enabling `serde` we are also enabling `std`. By defining a separate feature from `std`, we may be indicating to the users of crate that it is okay to enable composite feature `serde` without enabling `std` while in reality composite `serde` cannot be enabled without `std`. 

One way to resolve this is to remove this dependency on `std` here. But that would involve defining a no_std allocator and using `alloc::format` instead of `std::format`.